### PR TITLE
feat: Save unary operator on tracked entity filter [DHIS2-15954]

### PR DESCRIPTION
**Pull request by muilpp**
As with program stage working lists, I'm adding the null operator in tracked entity filters so it can be persisted to the database.

To validate the logic, I've created a new controller test, `TrackedEntityFilterControllerTest`, which replaces the `TrackedEntityFilterTest` end-to-end test.

Unrelated to this ticket, but while I'm at it, I'm using specific arguments in the change log test to create the page parameters, making the object creation explicit.

### CODE REVIEW

File: dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/JsonEntityQueryCriteria.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.trackedentityfilter;
+
+import org.hisp.dhis.jsontree.JsonArray;
+import org.hisp.dhis.jsontree.JsonObject;
+
+/** Representation of entity query criteria. */
+public interface JsonEntityQueryCriteria extends JsonObject {
```
Comment (ameenhere) [inline dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/JsonEntityQueryCriteria.java:34]: Why was this created?

File: dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/JsonTrackedEntityFilter.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.trackedentityfilter;
+
+import org.hisp.dhis.jsontree.JsonObject;
+
+/** Representation of a JSON object with entity query criteria. */
+public interface JsonTrackedEntityFilter extends JsonObject {
```
Comment (ameenhere) [inline dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/JsonTrackedEntityFilter.java:33]: Why was this created?

**Review by ameenhere — APPROVED** [review]

File: dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/TrackedEntityFilterControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+    assertFalse(trackedEntityFilter.isEmpty());
+    assertEquals(programId, trackedEntityFilter.getProgram());
+    JsonObject attributeObject =
+        trackedEntityFilter.getEntityQueryCriteria().getAttributeValueFilters().getObject(0);
+    assertEquals(trackedEntityAttributeId, attributeObject.getString("attribute").string());
+    assertEquals(Boolean.TRUE, attributeObject.getBoolean("null").booleanValue());
+  }
+
+  @Test
```
Comment (enricocolasante) [inline dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/TrackedEntityFilterControllerTest.java:205]: Would it make sense to add a test case when the value for the null operator is `null`?
The creation should fail, right?

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/TrackedEntityFilterControllerTest.java (hunk 3 above)
Comment (muilpp) [reply dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/TrackedEntityFilterControllerTest.java:205]: if the value is `null` it doesn't fail, and, regardless of the operator, it will only store the attribute.
as an example:

`"attributeValueFilters": [{"eq": null, "attribute": "ruQQnf6rswq"}]`
`"attributeValueFilters": [{"null": null, "attribute": "ruQQnf6rswq"}]`

both store only `{"attributeValueFilters": [{"attribute": "ruQQnf6rswq"}]}`
